src/test/java/redis/clients/jedis/tests/commands/SortedSetCommandsTest.java

```diff
@@ -542,7 +545,147 @@ public void zscore() {
     assertNull(bscore);
 
   }
+  
+  @Test
+  public void zpopmax() {
+    jedis.zadd("foo", 1d, "a");
+    jedis.zadd("foo", 10d, "b");
+    jedis.zadd("foo", 0.1d, "c");
+    jedis.zadd("foo", 2d, "d");
+
+    Tuple wrapper = jedis.zpopmax("foo");
+    String member = wrapper.getElement();
+    Double score = wrapper.getScore();
+    
+    assertEquals("b", member);
+    assertEquals((Double) 10d, score);
+
+    wrapper = jedis.zpopmax("foo");
+    member = wrapper.getElement();
+    score = wrapper.getScore();
+    
+    assertEquals("d", member);
+    assertEquals((Double) 2d, score);
+
+    wrapper = jedis.zpopmax("foo");
+    member = wrapper.getElement();
+    score = wrapper.getScore();
+    
+    assertEquals("a", member);
+    assertEquals((Double) 1d, score);
+    
+    wrapper = jedis.zpopmax("foo");
+    member = wrapper.getElement();
+    score = wrapper.getScore();
+    
+    assertEquals("c", member);
+    assertEquals((Double) 0.1d, score);
+    
+    // Empty
+    wrapper = jedis.zpopmax("foo");
+    assertTrue(wrapper == null);
+    
+    // Binary
+    jedis.zadd(bfoo, 1d, ba);
+    jedis.zadd(bfoo, 10d, bb);
+    jedis.zadd(bfoo, 0.1d, bc);
+    jedis.zadd(bfoo, 2d, ba);
+
+    // First
+    Tuple actual = jedis.zpopmax(bfoo);
+
+    Tuple expected = new Tuple(bb, 10d);
+    
+    assertEquals(expected, actual);
+
+    // Second
+    actual = jedis.zpopmax(bfoo);
+    expected = new Tuple(ba, 2d);
+    
+    assertEquals(expected, actual);
+    
+    // Third
+    actual = jedis.zpopmax(bfoo);
+    expected = new Tuple(bc, 0.1d);
+    
+    assertEquals(expected, actual);
+    
+    // Empty
+    actual = jedis.zpopmax(bfoo);
+    assertNull(actual);
+  }
+
+  @Test
+  public void zpopmaxWithCount() {
+    jedis.zadd("foo", 1d, "a");
+    jedis.zadd("foo", 10d, "b");
+    jedis.zadd("foo", 0.1d, "c");
+    jedis.zadd("foo", 2d, "d");
+    jedis.zadd("foo", 0.03, "e");
 
+    Set<Tuple> actual = jedis.zpopmax("foo", 2);
+    assertEquals(2, actual.size());
+    
+    Set<Tuple> expected = new HashSet<Tuple>();
+    expected.add(new Tuple("b", 10d));
+    expected.add(new Tuple("d", 2d));
+    
+    assertEquals(expected, actual);
+
+    actual = jedis.zpopmax("foo", 3);
+    assertEquals(3, actual.size()); // 3x [member, score]
+    
+    expected.clear();
+    
+    expected.add(new Tuple("a", 1d));
+    expected.add(new Tuple("c", 0.1d));
+    expected.add(new Tuple("e", 0.03d));
+    
+    assertEquals(expected, actual);
+    
+    // Empty
+    actual = jedis.zpopmax("foo", 1);
+    assertTrue(actual.isEmpty());
```

Comment:
Please keep one coding pattern for binary and non-binary within same method. Personally, I liked your code for binary better. E.g.
```
expected.clear();
assertEquals(expected, actual);
```